src/components/student-dashboard/progress-guide.cjsx

```diff
@@ -0,0 +1,146 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+LoadableItem = require '../loadable-item'
+_ = require 'underscore'
+S = require '../../helpers/string'
+
+{LearningGuideStore, LearningGuideActions} = require '../../flux/learning-guide'
+ChapterSection = require '../task-plan/chapter-section'
+ChapterSectionMixin = require '../chapter-section-mixin'
+
+ProgressGuide = React.createClass
+  displayName: 'ProgressGuide'
+
+  propTypes:
+    courseId: React.PropTypes.string.isRequired
+
+  mixins: [ChapterSectionMixin]
+
+  getRecentChapters: (chapter, i) ->
+    sections = _.map(chapter.children, @renderSectionBars)
+    <div key={i}>{sections}</div>
+
+  _percent: (num, total) ->
+    Math.round((num / total) * 100)
+
+  renderSectionBars: (section) ->
+    chapterSection =
+      @sectionFormat(section.chapter_section, @state.sectionSeparator)
+    sectionPercent = @_percent(section.current_level, 1)
+    colorClass = @colorizeBar(sectionPercent)
+    <div key={chapterSection} className='section'>
+      <div className='section-heading'>
+        <span className='section-number'>{chapterSection}</span>
+        <span className='section-title' title={section.title}>{section.title}</span>
+      </div>
+      <BS.ProgressBar className={colorClass} now={sectionPercent} />
+    </div>
+
+  colorizeBar: (percent) ->
```

Comment:
Should the progress bar rendering be in a common `.cjsx` file for both the dashboard widget and the Learning Guide?